server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -278,6 +279,7 @@ public void apply(Settings value, Settings current, Settings previous) {
                 FsRepository.REPOSITORIES_CHUNK_SIZE_SETTING,
                 FsRepository.REPOSITORIES_COMPRESS_SETTING,
                 FsRepository.REPOSITORIES_LOCATION_SETTING,
+                HotToWarmTieringService.WARM_TIERING_MAX_SHARD_SIZE,
```

Comment:
I was thinking about having a limit on max shard of an index to be tiered to warm such that when it is tiered back to hot from warm its a not unbounded download from the remote. However, that can also happen once the index is tiered to warm as the warm index is writable. So, I can remove this for now and this can be added later if needed.